torch/_inductor/lowering.py

```diff
@@ -1085,10 +1087,19 @@ def make_fallback(kernel, layout_constraint=None, warn=True):
     assert (
         kernel not in decompositions
     ), f"both a fallback and a decomp for same kernel: {kernel}"
-    if get_decompositions([kernel]) and warn:
-        developer_warning(
-            f"make_fallback({kernel}): a decomposition exists, we should switch to it"
+    if get_decompositions([kernel]) and warn and IS_CI:
+        # Note: 'warn' is holdover from when this was a warning, but for ops that previously
+        # set warn=False we do not want a CI error.
+        log.error(
+            "make_fallback(%s): a decomposition exists, we should switch to it."
+            " To fix this error, either add a decomposition to core_aten_decompositions (preferred)"
+            " or inductor_decompositions, and delete the corresponding `make_fallback` line."
+            " Get help from the inductor team if unsure, don't pick arbitrarily to unblock yourself.",
+            kernel,
         )
+        # Make sure the process exits and CI flags an error, raising
+        # an exception could allow a graph-break and silently pass CI in some configs
+        sys.exit(1)
```

Comment:
This seems cure is worse than disease. Why don't we raise an exception and write off the suppress error configs?

Reply:
I mean, I kinda knew you'd say that. But on balance, this way will stop fallbacks from getting checked in broken, and the only downside that I'm cognizant of is that it robs you of getting notified of further-down-the-stack failures on CI if it exits early.  I think it's slightly better to be sure not to allow bad things to land than to make devs rerun CI more times.  But ...

Reply:
No, I don't understand your response. What does `sys.exit(0)` give you that `raise AssertionError()` wouldn't?

Reply:
I am worried that dynamo will swallow exceptions sometimes in some tests in its attempt to gracefully fall back to eager. However, I hope there is some type of internal assertion we can raise that will reliably raise to userland, and if so i'm happy to use that instead.

Reply:
Perhaps we should do
```
if TEST_WITH_DYNAMO:
   sys.exit(1)
else:
   raise ....
```

Under `TEST_WITH_DYNAMO` raising will just fall back silently to eager and skip the real test.


Another option would be to dynamically switch `suppress_errors=False` and raise.